src/modality-agnostic-files/provenance.md

```diff
@@ -0,0 +1,485 @@
+   {
+      "Environments": "REQUIRED"
+   }
+) }}
+
+Each object in the `Environments` array includes the following keys:
+
+<!-- This block generates a table describing subfields within a metadata field.
+The definitions of these fields can be found in
+  src/schema/objects/metadata.yaml
+and a guide for using macros can be found at
+ https://github.com/bids-standard/bids-specification/blob/master/macros_doc.md
+-->
+{{ MACROS___make_subobject_table("metadata.Environments.items") }}
+
+!!! bug
+    TODO: Environment not currently defined in the context
+
+!!! example
+    Provenance metadata in a `prov/[<label>/]prov-<label>_ent.json` file:
+    ```JSON
+    {
+        "Environments": [
+            {
+                "Id": "bids::prov/#fedora-uldfv058",
+                "Label": "Fedora release 36 (Thirty Six)",
+                "OperatingSystem": "GNU/Linux 6.2.15-100.fc36.x86_64"
+            }
+        ]
+    }
+    ```
+    This snippet is an extract of the following comprehensive example: [Provenance records for DICOM to Nifti conversion using `dcm2niix`](https://github.com/bclenet/bids-examples/tree/BEP028_dcm2niix/provenance_dcm2niix)
+
+## Consistency of Ids
+
+The following rules and conventions are provided in order to have consistent, human readable, and explicit [IRIs](https://www.w3.org/TR/json-ld11/#iris) as `Id` for `Entity` provenance records.
+
+An `Id` identifying an `Entity` record corresponding to a file of a BIDS dataset MUST be a [BIDS URI](../common-principles.md#bids-uri).
+
+!!! Warning
+    The use of BIDS URIs may require to define the `DatasetLinks` object in [`dataset_description.json`](dataset-description.md#dataset_descriptionjson).
+
+!!! example "`Entity` naming examples"
+    - `bids:ds001734:sub-002/anat/sub-02_T1w.nii`: the `Id` of an `Entity` describing a T1w file for subject `sub-002` in the `ds001734` dataset ;
+    - `bids::sub-014/func/sub-014_task-MGT_run-01_events.tsv`: the `Id` of an `Entity` describing an events file for subject `sub-014` in the current dataset ;
+    - `bids:derivatives:fmriprep/sub-001/func/sub-001_task-MGT_run-01_bold_space-MNI152NLin2009cAsym_preproc.nii.gz`:  the `Id` of an `Entity` describing a bold file for subject `sub-001` in the `derivatives` dataset.
+
+An `Id` identifying `Activity`, `Software`, and `Environment` provenance records described in a BIDS dataset `<dataset>` SHOULD have the following form, where `<label>` is a human readable name for coherently identifying the record and `<uid>` is a unique group of chars:
+
+```text
+bids:<dataset>:prov#<label>-<uid>
+```
+
+!!! example "`Activity`, `Environment`, `Software` naming examples"
+    - `bids:ds001734:prov#conversion-xfMMbHK1`: a conversion `Activity` described inside the `ds001734` dataset;
+    - `bids::prov#fedora-uldfv058`: a Fedora based `Environment` described inside the current dataset.
+    - `bids:derivatives:prov#fmriprep-r4kzzMt8`: the fMRIPrep `Software` described inside the `derivatives` dataset.
```

Comment:
I feel we would need some very explicit statement about uniqueness of Ids, and enforce that in validator somehow.

```suggestion
    - `bids:derivatives:prov#fmriprep-r4kzzMt8`: the fMRIPrep `Software` described inside the `derivatives` dataset.

An `Id` must be unique, as to identify an instance and thus REQUIRED to be different across entries and different instances MUST have different `Id`s.
```